### PR TITLE
WebP support

### PR DESCRIPTION
**Pull request by adalberht**

### CODE REVIEW

File: pkg/processor/native/encoder.go [hunk 1]
```diff
@@ -35,6 +25,11 @@ type PngEncoder struct {
 	Encoder *png.Encoder
 }
 
+// WebPEncoder is an object to encode image to byte array with webp format
+type WebPEncoder struct {
+	Option *webp.Options
```
[the remark this answers is not in the document]
Comment (adalberht) [reply pkg/processor/native/encoder.go]: oh yes, missed that, thanks!

File: pkg/processor/native/encoder.go [hunk 2]
```diff
@@ -50,44 +45,77 @@ func (e *JpegEncoder) Encode(img image.Image) ([]byte, error) {
...
-	pngEncoder  Encoder
-	noOpEncoder Encoder
+	jpegEncoder *JpegEncoder
+	pngEncoder  *PngEncoder
+	noOpEncoder *NopEncoder
+	webPEncoder *WebPEncoder
 }
 
+// EncodersOption represents builder function for Encoders
+type EncodersOption func(*Encoders)
```
[the remark this answers is not in the document]
Comment (adalberht) [reply pkg/processor/native/encoder.go]: this one should be correct, Encoder is the single individual  object for encoding one specific extension, Encoders is a wrapper object for getting the object to encode the correct extension. Wdyt?

**Comment by adalberht** [comment]
Cool, thanks Ajat!

File: pkg/processor/native/encoder.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (ajatprabha) [reply pkg/processor/native/encoder.go]: Yes, missed that, makes sense then.

**Pull request merged**